program/rust/src/c_oracle_header.rs

```diff
@@ -58,6 +60,35 @@ impl PythAccount for PriceAccount {
     const INITIAL_SIZE: u32 = PC_PRICE_T_COMP_OFFSET as u32;
 }
 
+impl PythAccount for PermissionAccount {
+    const ACCOUNT_TYPE: u32 = PC_ACCTYPE_PERMISSIONS;
+    const INITIAL_SIZE: u32 = size_of::<PermissionAccount>() as u32;
+}
+
+/// This account stores the pubkeys that can execute administrative instructions in the Pyth
+/// program. Only the upgrade authority of the program can update these permissions.
+#[repr(C)]
+#[derive(Copy, Clone, Pod, Zeroable)]
+pub struct PermissionAccount {
+    /// pyth account header
+    pub header:                  AccountHeader,
+    /// An authority that can do any administrative task
+    pub master_authority:        Pubkey,
+    /// An authority that can  :
+    /// - Add mapping accounts
+    /// - Add price accounts
+    /// - Add product accounts
+    /// - Delete price accounts
+    /// - Delete product accounts
+    /// - Update product accounts
+    pub data_curation_authority: Pubkey,
+    /// An authority that can  :
+    /// - Add publishers
+    /// - Delete publishers
```

Comment:
(not a comment for now) I think we should think harder about the exact permissions given to these keys. I think add/remove/min publishers should belong to the data curation authority, because that is part of curating a data feed. The security authority should get the ability to temporarily depermission a small number of publishers or something ilke that.